docs/guides/building-forms.md

```diff
@@ -0,0 +1,387 @@
+
+Vercel empowers you to write JavaScript Serverless functions and deploy them at its edge network. This significantly minimizes latency because your web application runs code in a serverless architecture as close to the end-user as possible. It works incredibly well if you want to build lightweight and flexible applications that can be expanded or updated quickly.
+
+Having said that, let's jump on to the next part, where you'll be recreating these HTML forms in React using Next.js.
+
+Create a simple Next.js app. You can use the [create-next-app](https://nextjs.org/docs/getting-started#setup) for a quick start. In your command line terminal, run the following:
+
+```
+npx create-next-app
+```
+
+Answer the questions to create your project, and give it a name, this example uses [`next-forms`](https://github.com/vercel/next.js/tree/canary/examples/next-forms). Next `cd` into this directory, and run `npm run dev` or `yarn dev` command to start the development server.
+
+Open the URL printed in the terminal to ensure that your app is running successfully.
+
+## Setting up the Serverless Environment
+
+At the beginning of this guide, you learned that web forms have a client-server relationship. Now you will set up the server environment using Vercel.
+
+Vercel supports out-the-box deployments of [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions), which you can code in your favorite backend languages (Node.js, Go, Python and Ruby). These functions take an HTTP request and return a response.
+
+Serverless functions perform a significant role in handling tasks like form submission because:
+
+- Serverless functions are event-based, and every time you submit a form, it triggers a new event.
+- Offer faster deployments with greater flexibility (you don't have to manage any servers).
+- Reduce architecture and management costs.
+
+### Next.js Form Serverless API Endpoint
+
+Both the client and the server will be built using Next.js. For the server part, create a serverless API where you will send the form data.
+
+Next.js offers a file-based system for routing that's built on the [concept of pages](/docs/basic-features/pages). Any file inside the folder `pages/api` is mapped to `/api/*` and will be treated as an API endpoint instead of a page. This API endpoint is going to be server-side only.
```

Comment:
```suggestion
Next.js offers a file-based system for routing that's built on the [concept of pages](/docs/basic-features/pages). Any file inside the folder `pages/api` is mapped to `/api/*` and will be treated as an API endpoint instead of a page. This [API endpoint](/docs/api-routes/introduction) is going to be server-side only.
```